samplejava/src/main/java/com/example/chattutorialjava/ChannelActivity3.java

```diff
@@ -111,26 +109,28 @@ public void handleOnBackPressed() {
         typingHeader.setText(nobodyTyping);
 
         // Obtain a ChannelController
-        ChannelController channelController =
-                ChatDomain.instance().getUseCases()
-                        .getGetChannelController().invoke(cid)
-                        .execute().data();
-
-        // Observe typing users
-        channelController.getTyping().observe(this, users -> {
-            if (users.isEmpty()) {
-                typingHeader.setText(nobodyTyping);
-            } else {
-                StringBuilder typingText = new StringBuilder("typing: ");
-                for (int i = 0; i < users.size(); i++) {
-                    User user = users.get(i);
-                    if (i > 0) {
-                        typingText.append(", ");
+        ChatDomain.instance().getUseCases().getGetChannelController().invoke(cid).enqueue((result) -> {
+            ChannelController channelController = result.data();
+
+            runOnUiThread(() -> {
+                // Observe typing users
+                channelController.getTyping().observe(this, users -> {
+                    if (users.isEmpty()) {
+                        typingHeader.setText(nobodyTyping);
+                    } else {
+                        StringBuilder typingText = new StringBuilder("typing: ");
+                        for (int i = 0; i < users.size(); i++) {
+                            User user = users.get(i);
+                            if (i > 0) {
+                                typingText.append(", ");
+                            }
+                            typingText.append(user.getExtraData().get("name"));
+                        }
+                        typingHeader.setText(typingText.toString());
```

Comment:
We should replace with `TextUtils.join` too

Reply:
That's would be great, but unfortunately `TextUtils.join` joins an array of objects by calling `toString()` on each of them and it doesn't give a possibility to pass transform function. 
The problem is that we need to join user's name, so if we would like to use `TextUtils.join` we would need to map user's collection :(

Reply:
The code for mapping would be a lot nicer to write in Java than all this string concatenation, I think that would be worth doing, if we can then use `TextUtils.join`